components/tag/stories/tag.stories.js

```diff
@@ -49,7 +34,22 @@ export default {
 				category: "Content",
 			},
 			control: { type: "file", accept: ".svg,.png,.jpg,.jpeg,.webc" },
-			if: { arg: "hasAvatar", truthy: true },
+			if: { arg: "visual", eq: "Avatar" },
+		},
+		visual: {
+			name: "Visual",
+			type: { name: "string" },
+			table: {
+				type: { summary: "string" },
+				category: "Component",
+			},
+			options: [
+				"None",
+				"Avatar",
+				"Icon",
+				"Thumbnail",
+			],
+			control: "select",
```

Comment:
Nice work combining these into a single control!
It might be worth also adding a description to this, like "The type of visual element displayed before the label".